lipstick/animations.py

```diff
@@ -25,45 +25,52 @@ def update_fig(fig, axes, sleep=0):
 
 
 class GifMaker:
-    def __init__(self, filename, fps=30.0):
+    def __init__(self, filename, fps=30.0, dpi=100, temp_dir_name="gifmaker_cache", keep_dir=False,):
         """
         Saves gif from added figures.
 
         Args:
             filename (str): full path and filename of the resulting gif (e.g. "sample.gif")
             fps (float, optional): Frames per second. Defaults to 30.0.
+            dpi (int, optional): Dots per inch of each image. Defaults to 100. Choose 150 for high quality gifs.
+            temp_dir_name (str, optional): Name of the temporary directory. Defaults to "gifmaker_cache".
+            keep_dir (bool, optional): If True, the temporary directory is not deleted. Useful when trying to create videos with custom ffmpeg commands. Defaults to False.
+
+        How to convert the gif to a video:
+            - set keep_dir=True
+            - after all the images are saved, navigate to the temporary directory
+            - make sure ffmpeg is installed on your system
+            - run the following command within the temporary directory:
+
+                ffmpeg -framerate 30 -pattern_type glob -i '*.png' -c:v libx264 out.mp4
+
+            - the video will be saved as "out.mp4" in the temporary directory, with 30 fps.
         """
 
         self.path = "/".join(filename.split(".")[0].split("/")[:-1])
         self.path = self.path if self.path else os.getcwd()
         self.name = filename.split(".")[0].split("/")[-1]
         self.fps = fps
-        self.temp_dir = "gifmaker_cache"
+        self.dpi = dpi
+        self.temp_dir = temp_dir_name
         self.i = 0
+        self.keep_dir = keep_dir
 
     def add(self, fig):
         fig.savefig(
             f"{self.temp_dir}/{self.name}" + str(self.i).zfill(4) + ".png",
             bbox_inches="tight",
+            dpi=self.dpi,
```

Comment:
Why not set this when creating the figure? Like `fig = plt.figure(dpi=100)` or `fig, ax = plt.subplots(dpi=100)`.
Both options are possible, and that way we do not intervene with how the user wants their figure to look like and it is their own responsibility.

Reply:
yes, I agree.
I changed the default of dpi in the GifMaker, so that the the `dpi` of the figure is used per default. But it can be overwritten if desired. I can also remove it entirely if you think that's best.